### DISCUSSION TITLE
Resolve parsing ambiguity

### DISCUSSION THREAD

**Discussion by shaded-enmity** [tree-sitter/tree-sitter#1877] [Q&A] [answered]
Hello tree-sitter community!

I'm working on a tree-sitter grammar for the Rego language, while my work is mostly done, I've just discovered a bug in the grammar and need a second pair of eyes because I'm kinda stuck.

The grammar in question can be found here: FallenAngel97/tree-sitter-rego#2

The ambiguity can be seen here:

```
test := true
^^^^    ^^^^ rule_body
 rule_head
```

`test` is `rule_head` and `true` is `rule_body`

```
test := bar { bar := "test" }
^^^^^^^^^^^ ^^^^^^^^^^^^^^^^^ rule_body
 rule_head
```

`test := bar` is `rule_head` and `{ bar := "test" }` is `rule_body`

---

Couple of questions:
1) Is it possible to do a lookahead? i.e. parse things differently when `{` is encountered while processing `rule_head`? 
2) Do I need to write C code to deal with this?
3) Am I missing some obvious way how to resolve this ambiguity?

**Comment by ahelwer** [comment, marked as answer]
What happens if you write up the rules to parse those things then declare a conflict so GLR parsing is used?

**Reply by shaded-enmity** [reply]
Hmm, let me see if I understand: I need to update my grammar so that `tree-sitter generate` doesn't work without `conflicts: [[$.rule_head, $.rule_body]]` in `grammar.js`?

**Reply by ahelwer** [reply]
Correct
